hooks/pre-command

```diff
@@ -4,15 +4,29 @@ set -euo pipefail
 
 DIR="$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)"
 
-# shellcheck disable=SC1091
-. "${DIR}/../lib/setup.bash"
-
-GH_TOKEN="${GH_TOKEN:-${GITHUB_TOKEN:-}}"
+OBLT_CLI_GH_TOKEN="${GH_TOKEN:-${GITHUB_TOKEN:-${VAULT_GITHUB_TOKEN:-}}}"
 OBLT_CLI_USERNAME="${BUILDKITE_PLUGIN_OBLT_CLI_USERNAME:-obltmachine}"
 OBLT_CLI_SLACK_CHANNEL="${BUILDKITE_PLUGIN_OBLT_CLI_SLACK_CHANNEL:-#observablt-bots}"
 OBLT_CLI_VERSION="${BUILDKITE_PLUGIN_OBLT_CLI_VERSION:-}"
 OBLT_CLI_VERSION_FILE="${BUILDKITE_PLUGIN_OBLT_CLI_VERSION_FILE:-"${DIR}/../.default-oblt-cli-version"}"
 
+if [[ -z ${OBLT_CLI_GH_TOKEN} ]]; then
+	>&2 echo "The GH_TOKEN, GITHUB_TOKEN or VAULT_GITHUB_TOKEN environment variable must be set."
+	exit 1
+fi
+
+# shellcheck disable=SC1091
+. "${DIR}/../lib/setup.bash"
+
+git config --global user.name "${GIT_USER:-"${OBLT_CLI_USERNAME}"}"
+git config --global user.email "${GIT_EMAIL:-"${OBLT_CLI_USERNAME}@users.noreply.github.com"}"
+
+git config --global credential.helper store
+echo "protocol=https
+host=github.com
+username=token
+password=$OBLT_CLI_GH_TOKEN" | git credential-store store
```

Comment:
Should a pre-exit hook be included too and delete this entry? Just wanna be sure static workers don't have secrets in plain text

Reply:
https://github.com/elastic/oblt-cli-buildkite-plugin/pull/7/commits/ab0d60ca88d616bcfb2025f25571b00294f92be5

Reply:
I don't know if you need this, I've been searching how the existing Buildkite/Vault integration works when provisioning BK runners:

- see [this](https://github.com/elastic/vault-buildkite-plugin/blob/d2f85f85efdf229b341c3653938a0c64d792a6a5/hooks/environment#L314) (Only accessible by  Elastic employees)

Reply:
You mean, we can revert the commit?

Reply:
I think so, the existing BK/Vault integration configures the username/password using the GitHub ephemeral token. And if we change it then it might cause some issues?

I wonder whether  the `OBLT_CLI_GH_TOKEN` might not be needed at all and and we will need to rely on the `VAULT_GITHUB_TOKEN` and the permissions-set configuration, what do you think?

Reply:
The limitation of the permission set worries me. Because of oblt-cli, we need to change the `contents: read` permission to `contents: write`.

But I agree that if we only rely on the `VAULT_GITHUB_TOKEN`, it will have the most native experience and no surprises because of the modified git credentials.

Reply:
Revert commit: https://github.com/elastic/oblt-cli-buildkite-plugin/pull/7/commits/79ec172945d4a7fb7999ce2098951f00b025612c

Reply:
Updated in https://github.com/elastic/oblt-cli-buildkite-plugin/pull/9